packages/terra-date-picker/src/DateInput.jsx

```diff
@@ -0,0 +1,38 @@
+import React, { PropTypes } from 'react';
+import Button from 'terra-button';
+import IconCalendar from 'terra-icon/lib/icon/IconCalendar';
+
+const Icon = <IconCalendar />;
+
+// eslint-disable-next-line react/prefer-stateless-function
+class DatePickerInput extends React.Component {
```

Comment:
I initialize made this functional but I started seeing this warning in the console
> Warning: Stateless function components cannot be given refs (See ref "input" in 
> DatePickerInput created by DateInput). Attempts to access this ref will fail.
I think it is referring to this [ref](https://github.com/Hacker0x01/react-datepicker/blob/master/src/datepicker.jsx#L353) in react-datepicker. So I made it a class to get rid of the warning.